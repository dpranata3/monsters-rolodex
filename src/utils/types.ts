export interface Monster {
  email: string;
  id: number;
  name: string;
}