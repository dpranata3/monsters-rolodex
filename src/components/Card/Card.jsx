import React from "react";
import styled from "styled-components";

const Card = ({ monster }) => {
  const { id, name, email } = monster;
  return (
    <CardContainer>
      <img alt={name} src={`https://robohash.org/${id}/set=3&size=180x180`} />
      <h2>{name}</h2>
      <p>{email}</p>
    </CardContainer>
  );
};

export default Card;

const CardContainer = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #95dada;
  border: 1px solid grey;
  border-radius: 5px;
  padding: 25px;
  cursor: pointer;
  -moz-osx-font-smoothing: grayscale;
  backface-visibility: hidden;
  transform: translateZ(0);
  transition: transform 0.25s ease-out;

  :hover {
    transform: scale(1.05);
    background-color: #f0f728;
  }
`;
