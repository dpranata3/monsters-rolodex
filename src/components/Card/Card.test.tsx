import { shallow } from "enzyme";

import Card from "./Card";
import { Monster } from "../../utils/types";

const mockMonsterData: Monster = {
  id: 1,
  name: "Mocking Bird",
  email: "mockme@mail.com",
};

it("expect to render card component with monster data as props", () => {
  expect(shallow(<Card monster={mockMonsterData} />)).toMatchSnapshot();
});
