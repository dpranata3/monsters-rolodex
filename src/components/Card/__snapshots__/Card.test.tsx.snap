// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component with monster data as props 1`] = `ShallowWrapper {}`;
